docs/develop/gettingstarted.rst

```diff
@@ -0,0 +1,281 @@
+   or any other sources on the internet.
+
+Get the code
+----------------
+
+It is recommended that you maintain a fork of the MicroPython repository for your development purposes.
+The process of obtaining the source code includes the following:
+
+#. Fork the repository https://github.com/micropython/micropython.git
+#. You will now have a fork at <https://github.com/<yourusername>/micropython.git>.
+#. Clone the repository. You may clone either your fork or the main MicroPython repository.
+   Use one of the commands below:
+
+.. code-block:: console
+
+   $ git clone https://github.com/micropython/micropython.git 
+   $ git clone https://github.com/<yourusername>/micropython.git
+
+Depending on what you cloned, do the necessary configuration.
+
+Configure upstream if you cloned your fork.
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+
+Configure upstream and origin if you cloned the main repository. You can maintain your
+own mapping but it is recommended that origin maps to your fork and upstream to the main
+MicroPython repository:
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+   $ git remote set-url origin https://github.com/<yourusername>/micropython.git
+
+After the above configuration, your setup should be similar to this:
+
+.. code-block:: console
+
+   origin	https://github.com/<yourusername>/micropython.git (fetch)
+   origin	https://github.com/<yourusername>/micropython.git (push)
+   upstream	https://github.com/micropython/micropython.git (fetch)
+   upstream	https://github.com/micropython/micropython.git (push)
+
+You should now have a copy of the source code. By default you are pointing
+to the master branch. To get ready for further development, it is recommended
+to work on a development branch.
+
+.. code-block:: console
+
+    $ git checkout -b dev-branch
+
+You can give it any name. You will have to compile MicroPython whenever you change 
+to a different branch.
+
+Build the Code
+--------------
+
+For development, you should compile MicroPython for a specific platform which corresponds
+to a port. Before any building for a port, first build the MicroPython cross-compiler.
+Install the required dependencies for the port in order to sucessfully
+compile and build.
+
+Build the cross-compiler
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Almost all ports require building ``mpy-cross`` first to perform pre-compilation
+of scripts that will be included in the port firmware:
+
+.. code-block:: console
+
+   $ cd mpy-cross
+   $ make
+
+If ``mpy-cross`` built successfully, you should see a message similar to this:
+
+.. code-block:: console
+
+   LINK mpy-cross
+      text	   data	    bss	    dec	    hex	filename
+    279328	    776	    880	 280984	  44998	mpy-cross
+
+The unix port
```

Comment:
We may need to provide some guidance on what the unix port is. I imagine many beginners wouldn't know if they need to build it or not! 
The description of what the unix port is doesn't necessarily have to be _here_ though. A link would be fine as long as it will help a user understand if they need to build the port or not.